tutorials-v5/time-evolution/023_v5_paper-mesolve.md

```diff
@@ -0,0 +1,565 @@
+This new interface is class-based, allowing users to instantiate a solver object for a specific problem.
+This can be useful when the same Hamiltonian data is reused with different initial conditions, time steps or other options.
+A noticable speed-up can thus be achieved if solvers are reused multiple times.
+
+Upon instantiation, one first supplies only the Hamiltonian and the collapse operators (e.g., collapse operators for a Lindbladian master equation).
+Initial conditions, time steps, etc. are passed to the `Solver.run()` method which then performs the simulation.
+
+In this notebook we will consider several examples illustrating the usage of the new solver classes.
+
+```python
+import matplotlib.pyplot as plt
+import numpy as np
+from qutip import (QobjEvo, SESolver, UnderDampedEnvironment,
+                   about, basis, brmesolve, fidelity, mesolve,
+                   qeye, sigmam, sigmax, sigmaz, spost, spre, sprepost)
+
+from qutip.solver.heom import HEOMSolver
+
+%matplotlib inline
+```
+
+## Part 0: Introduction to the New Solver Class
+
+In our first example, we want to look at two interacting qubits that are (for now) decoupled from an environment.
+Such a system is described by the Hamiltonian
+
+$H = \dfrac{\epsilon_1}{2} \sigma_z^{(1)} + \dfrac{\epsilon_2}{2} \sigma_z^{(2)} + g \sigma_{x}^{(1)} \sigma_{x}^{(2)}$.
+
+The Pauli matrices $\sigma_z^{(1/2)}$ describe the respective two-level system of the qubits.
+The qubits are coupled via $\sigma_{x}^{(1/2)}$ and their interaction strength is given by $g$.
+
+```python
+epsilon1 = 1.0
+epsilon2 = 1.0
+g = 0.1
+
+sx1 = sigmax() & qeye(2)
+sx2 = qeye(2) & sigmax()
+sz1 = sigmaz() & qeye(2)
+sz2 = qeye(2) & sigmaz()
+
+H = (epsilon1 / 2) * sz1 + (epsilon2 / 2) * sz2 + g * sx1 * sx2
+
+print(H)
+```
+
+The dynamics of such a system is described by the Schrödinger equation
+
+$i \hbar \dfrac{d}{dt} \ket{\psi} = H \ket{\psi}$.
+Therefore, we can use `SESolver` to calculate the dynamics.
+
+```python
+se_solver = SESolver(H)
+psi0 = basis(2, 0) & basis(2, 1)
+tlist = np.linspace(0, 40, 100)
+```
+
+```python
+se_res = se_solver.run(psi0, tlist, e_ops=[sz1, sz2])
+```
+
+```python
+plt.plot(tlist, se_res.expect[0], label="i=1")
+plt.plot(tlist, se_res.expect[1], label="i=2")
+plt.xlabel("Time")
+plt.ylabel(r"$\langle \sigma_z^{(i)} \rangle$")
+plt.legend()
+plt.show()
+```
+
+### Manual Stepping Interface
+
+A new feature in QuTiP v5 is that time steps can be controlled manually.
+This is specifically useful if the Hamiltonian depends on an external control parameter such as a field strength.
+Such parameters can be updated in each step using the optional parameter `args`.
+In practice, this can look like this:
+
+```python
+t = 0
+dt = 40 / 100
+se_solver2 = SESolver(H)
+se_solver2.start(psi0, t)
+while t < 40:
+    t = t + dt
+    psi = se_solver2.step(t)
+    # process the result psi and calculate next time step
+```
```

Comment:
This example is a bit meh to be honest (it is from me 🙂), and it doesn't really show the `args` stuff that the text above it discusses... Not sure what to do about it